rhino/src/main/java/org/mozilla/javascript/IdScriptableObject.java

```diff
@@ -922,62 +932,96 @@ protected ScriptableObject getOwnPropertyDescriptor(Context cx, Object id) {
         ScriptableObject desc = super.getOwnPropertyDescriptor(cx, id);
         if (desc == null) {
             if (id instanceof String) {
-                return getBuiltInDescriptor((String) id);
+                return getBuiltInDataDescriptor((String) id);
             }
 
             if (ScriptRuntime.isSymbol(id)) {
                 if (id instanceof SymbolKey) {
-                    return getBuiltInDescriptor((SymbolKey) id);
+                    return getBuiltInDataDescriptor((SymbolKey) id);
                 }
 
-                return getBuiltInDescriptor(((NativeSymbol) id).getKey());
+                return getBuiltInDataDescriptor(((NativeSymbol) id).getKey());
             }
         }
         return desc;
     }
 
-    private ScriptableObject getBuiltInDescriptor(String name) {
-        Object value = null;
-        int attr = EMPTY;
+    private Slot queryOrFakeSlot(Context cx, Object id) {
+        var slot = querySlot(cx, id);
+        if (slot == null) {
+            if (id instanceof String) {
+                return getBuiltInSlot((String) id);
+            }
 
+            if (ScriptRuntime.isSymbol(id)) {
+                if (id instanceof SymbolKey) {
```

Comment:
Also, looking at coverage, nothing from line 957 to 961 seems to get covered either.

Reply:
This one is a little more convoluted: as of right now, I think it's impossible for this code path to be executed, thus it will be challenging to test it. This is because it's only called from  `IdScriptableObject::defineOwnProperty` ( [here](https://github.com/nabacg/rhino/blob/e20ecb059baac35e24541fb78daf5fd0aa31d312/rhino/src/main/java/org/mozilla/javascript/IdScriptableObject.java#L870) and [here](https://github.com/nabacg/rhino/blob/e20ecb059baac35e24541fb78daf5fd0aa31d312/rhino/src/main/java/org/mozilla/javascript/IdScriptableObject.java#L897)) and before the call, we've already [established](https://github.com/mozilla/rhino/blob/e20ecb059baac35e24541fb78daf5fd0aa31d312/rhino/src/main/java/org/mozilla/javascript/IdScriptableObject.java#L861) that `key` must be a CharSequence, so it can't a Symbol.  So maybe the solutions is to remove this branch?

Having said that, this doesn't feel right and perhaps it's an issue with current implementation of `IdScriptableObject::defineOwnProperty`, since it's only handling String (or CharSequence) keys and essentially delegating Symbol key'ed properties to it's base class (ScriptableObject).  It's very consistent with how `getOwnPropertyDescriptor` implementation, which does try to find built in descriptor using both Strings and Symbols ([here](https://github.com/mozilla/rhino/pull/1746/files#diff-4fda7f3359a263960b860746308e275c0ec0e9e78462481905066c97eb5a69a2L930) ). I believe this currently works mostly, because we always check base class for property descriptor first, but it's a little confusing why we need that symbol lookup.  

One could also imagine, someday in the future, someone creating a new built-in `NativeX` class extending `IdScriptableObject` that defines a Symbol keyed property, that's shouldn't be redefined  ( i.e. `configurable: false` ) and current implementation of `IdScriptableObject::defineOwnProperty` would not check it ( only Strings are handled) and allow base class implementation to override it. 

I appreciate it's a little theoretical, but wanted to share this and ask for opinions on whether we should address this? Also if addressing this deserves a separate PR or should I add it here?

Reply:
OK, thanks -- I suppose one way we could deal with this could be to assert, and in the future if someone wants to implement a subclass that needs it, we'd fix it -- although in many cases we are trying to get rid of IdScriptableObject as it doesn't yield the performance benefit it once did. If you're not comfortable with that, however, I'd also be OK leaving this as it is.

Reply:
I personally wouldn't mind leaving it as is for now, I don't see an obvious place to assert that wouldn't break existing functionality or be completely redundant.

Reply:
OK, then I'm satisfied, thanks!